src/jace/stages.py

```diff
@@ -0,0 +1,333 @@
+            - Support keyword arguments and default values of the wrapped function.
+            - Support static arguments.
+        """
+        call_args = tuple(tcache._AbstractCallArgument.from_value(x) for x in args)
+        return tcache.StageTransformationSpec(stage_id=id(self), call_args=call_args)
+
+
+class JaCeLowered(tcache.CachingStage["JaCeCompiled"]):
+    """Represents the original computation as an SDFG.
+
+    This class represents the output of `JaCeWrapped.lower()` and represents the originally wrapped
+    computation as an SDFG. This stage is followed by the `JaCeCompiled` stage.
+
+    Args:
+        tsdfg:      The translated SDFG object representing the computation.
+
+    Note:
+        `self` will manage the passed `tsdfg` object. Modifying it results in undefined behavior.
+        Although, `JaCeWrapped` is composable with Jax transformations `JaCeLowered` is not.
+        A user should never create such an object, instead `JaCeWrapped.lower()` should be used.
+
+    Todo:
+        - Handle pytrees.
+    """
+
+    _translated_sdfg: translator.TranslatedJaxprSDFG
+
+    def __init__(
+        self,
+        tsdfg: translator.TranslatedJaxprSDFG,
+    ) -> None:
+        super().__init__()
+        self._translated_sdfg = tsdfg
+
+    @tcache.cached_transition
+    def compile(
+        self,
+        compiler_options: CompilerOptions | None = None,
+    ) -> JaCeCompiled:
+        """Optimize and compile the lowered SDFG using `compiler_options`.
+
+        Returns an object that encapsulates a compiled SDFG object. To influence the various
+        optimizations and compile options of JaCe you can use the `compiler_options` argument.
+        If nothing is specified `jace.optimization.DEFAULT_OPTIMIZATIONS` will be used.
+
+        Note:
+            Before `compiler_options` is forwarded to `jace_optimize()` it will be merged with
+            the default arguments.
+        """
+        # We **must** deepcopy before we do any optimization, because all optimizations are in
+        #  place, however, to properly cache stages, they have to be immutable.
+        tsdfg: translator.TranslatedJaxprSDFG = copy.deepcopy(self._translated_sdfg)
+        optimization.jace_optimize(tsdfg=tsdfg, **self._make_compiler_options(compiler_options))
+
+        return JaCeCompiled(
+            csdfg=util.compile_jax_sdfg(tsdfg),
+            inp_names=tsdfg.inp_names,
+            out_names=tsdfg.out_names,
+        )
+
+    def compiler_ir(self, dialect: str | None = None) -> translator.TranslatedJaxprSDFG:
+        """Returns the internal SDFG.
+
+        The function returns a `TranslatedJaxprSDFG` object. It is important that modifying this
+        object in any way is undefined behavior.
+        """
+        if (dialect is None) or (dialect.upper() == "SDFG"):
+            return self._translated_sdfg
+        raise ValueError(f"Unknown dialect '{dialect}'.")
+
+    def as_html(self, filename: str | None = None) -> None:
```

Comment:
I see you point, but these kind of functions in Jax are all named `as_something()` thus we keep them.

Reply:
But the key is that this function is not the same kind of function because it's not returning a value but running a side effect. The current `as_html()` name doesn't make sense and I still think it should be changed.

Reply:
You are right, I now named it plain `view()`.